graphqlapi/src/test/java/io/stargate/graphql/schema/schemafirst/fetchers/admin/SchemasPerNamespaceFetcherTest.java

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.fetchers.admin;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import graphql.schema.DataFetchingEnvironment;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthenticationSubject;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.schema.Schema;
+import io.stargate.graphql.persistence.schemafirst.SchemaSource;
+import io.stargate.graphql.persistence.schemafirst.SchemaSourceDao;
+import io.stargate.graphql.schema.schemafirst.util.Uuids;
+import java.util.*;
+import org.junit.jupiter.api.Test;
+
+class SchemasPerNamespaceFetcherTest {
+
+  @Test
+  public void getSchemasPerNamespaces() throws Exception {
+    // given
+    String namespace = "ns_1";
+    SchemaSource schemaSource1 = new SchemaSource(namespace, Uuids.timeBased(), "content");
+    SchemaSource schemaSource2 = new SchemaSource(namespace, Uuids.timeBased(), "content-2");
+
+    SchemaSourceDao schemaSourceDao = mock(SchemaSourceDao.class);
+    when(schemaSourceDao.getSchemaHistory(namespace))
+        .thenReturn(Arrays.asList(schemaSource1, schemaSource2));
+    SchemasPerNamespaceFetcher schemasPerNamespaceFetcher =
+        new SchemasPerNamespaceFetcher(
+            mock(AuthenticationService.class),
+            mock(AuthorizationService.class),
+            mock(DataStoreFactory.class),
+            (ds) -> schemaSourceDao);
+
+    DataFetchingEnvironment dataFetchingEnvironment = mockDataFetchingEnvironment(namespace);
+    DataStore dataStore = mockDataStore(namespace);
+
+    // when
+    List<Map<String, Object>> result =
+        schemasPerNamespaceFetcher.get(
+            dataFetchingEnvironment, dataStore, mock(AuthenticationSubject.class));
+
+    // then
+    assertThat(result).hasSize(2);
+    Map<String, Object> firstSchema = result.get(0);
+    assertThat(firstSchema)
+        .contains(
+            new AbstractMap.SimpleEntry<>("namespace", namespace),
+            new AbstractMap.SimpleEntry<>("version", schemaSource1.getVersion()),
+            new AbstractMap.SimpleEntry<>("deployDate", schemaSource1.getDeployDate()));
+    Map<String, Object> secondSchema = result.get(1);
+    assertThat(secondSchema)
+        .contains(
+            new AbstractMap.SimpleEntry<>("namespace", namespace),
+            new AbstractMap.SimpleEntry<>("version", schemaSource2.getVersion()),
+            new AbstractMap.SimpleEntry<>("deployDate", schemaSource2.getDeployDate()));
```

Comment:
Checking the raw result like this is a bit tedious. It would look better if these tests executed GraphQL queries.
I think we have all the pieces already, `GraphQlTestBase` has the infrastructure to load a schema and run it against a mock datastore. `ScalarsDmlTest#shouldEncodeScalarValues` shows how to mock a result set.

If you don't want to sidetrack the feature we can create a follow-up ticket.

Reply:
I agree that `GraphQlTestBase` test adds using the graphql schema, but it also mocks everything underneath.
I wonder that maybe we should take a more fully IT approach when testing graph first: 
-use the underlying persistence via ccm
- execute queries via graphql rest API
- validate the data in the actual persistence
It may give us a lot more confidence. Regardless of the approach we take, I think we should implement (and discuss) that in the follow-up ticket to cover more graphql-first code. I am happy to work on it.